### PR TITLE
test(amplify_authenticator): signup with phone

### PR DESCRIPTION
**Pull request by haverchuck**
*Description of changes:*
Adds signup with phone tests.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/amplify_authenticator/example/pubspec.yaml [hunk 1]
```diff
@@ -21,6 +21,7 @@ environment:
   sdk: ">=2.12.0 <3.0.0"
 
 dependencies:
+  amplify_api: ^0.2.0
```
Comment (dnys1) [inline packages/amplify_authenticator/example/pubspec.yaml:24]: dev dependency

File: packages/amplify_authenticator/example/integration_test/utils/mock_data.dart [hunk 2]
```diff
@@ -11,6 +26,7 @@ const digits = '1234567890';
 const symbols = '~/`!@#\$%^&\\"\'*(),._?:;{}|<>';
 const String mockPhoneNumber = '+15555551234';
 const String mockCode = '12345';
+const String getPhoneQuery = '''query GetPhone {getPhone {phone}}''';
```
Comment (dnys1) [inline packages/amplify_authenticator/example/integration_test/utils/mock_data.dart:29]: What's this for?

File: packages/amplify_authenticator/example/integration_test/sign_up_with_phone_test.dart [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+      // And I type my "email" with a randomized email
+      await signUpPage.enterEmail(email);
+
+      // And I click the "Create Account" button
+      await signUpPage.submitSignUp();
+
+      // Then I see "Confirmation Code"
+      await confirmSignUpPage.expectConfirmSignUpIsPresent();
+      confirmSignUpPage.expectConfirmationCodeIsPresent();
+      await deleteUser(phone);
```
Comment (dnys1) [inline packages/amplify_authenticator/example/integration_test/sign_up_with_phone_test.dart:106]: I would put this in `tearDownAll`

File: packages/amplify_authenticator/example/integration_test/sign_up_with_phone_test.dart [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+  group('sign-up-with-phone', () {
+    // Given I'm running the example "ui/components/authenticator/sign-up-with-username"
+    setUpAll(() async {
+      await loadConfiguration(
+        'ui/components/authenticator/sign-up-with-phone',
+        additionalConfigs: [AmplifyAPI()],
+      );
+    });
+
+    // Scenario: Login mechanism set to "email"
```
Comment (dnys1) [inline packages/amplify_authenticator/example/integration_test/sign_up_with_phone_test.dart:53]: typo

```suggestion
    // Scenario: Login mechanism set to "phone_number"
```

File: packages/amplify_authenticator/example/integration_test/sign_up_with_phone_test.dart [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+    testWidgets('Login mechanism set to "phone"', (tester) async {
+      SignUpPage signUpPage = SignUpPage(tester: tester);
+      SignInPage signInPage = SignInPage(tester: tester);
+      await loadAuthenticator(tester: tester, authenticator: authenticator);
+      await signInPage.navigateToSignUp();
+      signUpPage.expectUserNameIsPresent(usernameLabel: 'Phone Number');
```
Comment (dnys1) [inline packages/amplify_authenticator/example/integration_test/sign_up_with_phone_test.dart:59]: Missing `And I don't see "Username" as an input field`

File: packages/amplify_authenticator/example/integration_test/sign_up_with_phone_test.dart (hunk 3 above)
Comment (haverchuck) [reply packages/amplify_authenticator/example/integration_test/sign_up_with_phone_test.dart:114]: The to-be-deleted user is only created on this final test.

Comment (dnys1) [reply packages/amplify_authenticator/example/integration_test/sign_up_with_phone_test.dart:114]: Ah, my mistake. Sounds good

**Review by dnys1 — APPROVED** [review]

**Review by Jordan-Nelson — APPROVED** [review]

**Pull request merged**